src/libcore/slice.rs

```diff
@@ -1185,6 +1185,40 @@ impl<'a, T> Iterator for Windows<'a, T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<'a, T> DoubleEndedIterator for Windows<'a, T> {
+    #[inline]
+    fn next_back(&mut self) -> Option<&'a [T]> {
+        if self.size > self.v.len() {
+            None
+        } else {
+            let ret = Some(&self.v[self.v.len()-self.size..]);
+            self.v = &self.v[..self.v.len()-1];
+            ret
+        }
+    }
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<'a, T> ExactSizeIterator for Windows<'a, T> {}
```

Comment:
This looks like it may not actually be able to be implemented due to this line in `size_hint`:

```rust
(x.saturating_add(1), x.checked_add(1))
```

If `x` were `uint::MAX` then this would return `(uint::MAX, None)`, breaking the contract of `ExactSizeIterator`. The value of `x`, however is `self.v.len() - self.size` and we know that `size != 0` so we're guaranteed that the `checked_add` will always return `Some`.

Could you update the above `size_hint` to return `(x + 1, Some(x + 1))` to be clear that it is a precise size?